From 3a14b397e0af5c80702ab9246806837d830260e8 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Tue, 13 Nov 2012 11:12:15 +0800
Subject: [PATCH 367/498] ENGR00232178 compile: avoid using relative path

1.In android module's Makefile, should avoid using a relative path.
Or it will refer to a wrong path in -jN parallel compile.
2.use ccflags-y instead of CFLAGS out of kernel to avoid compile error
This is supplement for:
ENGRxxxxx wifi: fix parallel compile error

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 wlan/atheros/compat-wireless/Android.mk |  4 ++--
 wlan/atheros/compat-wireless/Makefile   | 15 ++++++++-------
 2 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/wlan/atheros/compat-wireless/Android.mk b/wlan/atheros/compat-wireless/Android.mk
index 243cba6..ed2497c 100644
--- a/wlan/atheros/compat-wireless/Android.mk
+++ b/wlan/atheros/compat-wireless/Android.mk
@@ -37,13 +37,13 @@ ATH_CROSS_COMPILE=$(ATH_ANDROID_ROOT)/prebuilts/gcc/linux-x86/arm/arm-eabi-4.6/b
 mod_cleanup := $(ATH_ANDROID_ROOT)/$(ATH_ANDROID_SRC_BASE)/dummy
 
 $(mod_cleanup) :
-	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) ARCH=arm CROSS_COMPILE=$(ATH_CROSS_COMPILE) KLIB=$(ATH_LINUXPATH) KLIB_BUILD=$(ATH_LINUXPATH) clean
+	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) ARCH=arm CROSS_COMPILE=$(ATH_CROSS_COMPILE) KLIB=$(ATH_LINUXPATH) KLIB_BUILD=$(ATH_LINUXPATH) COMPAT_CURDIR=$(ATH_ANDROID_ROOT)/$(ATH_ANDROID_SRC_BASE) clean
 	mkdir -p $(TARGET_OUT)/etc/firmware/ath6k/AR6003/hw2.1.1/
 	mkdir -p $(TARGET_OUT)/lib/modules/
 
 ath6kl_module_file :=drivers/net/wireless/ath/ath6kl/ath6kl_sdio.ko
 $(ATH_ANDROID_SRC_BASE)/$(ath6kl_module_file):$(mod_cleanup) $(TARGET_PREBUILT_KERNEL) $(ACP)
-	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) O=$(ATH_LINUXPATH) ARCH=arm CROSS_COMPILE=$(ATH_CROSS_COMPILE) KLIB=$(ATH_LINUXPATH) KLIB_BUILD=$(ATH_LINUXPATH)
+	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) O=$(ATH_LINUXPATH) ARCH=arm CROSS_COMPILE=$(ATH_CROSS_COMPILE) KLIB=$(ATH_LINUXPATH) KLIB_BUILD=$(ATH_LINUXPATH) COMPAT_CURDIR=$(ATH_ANDROID_ROOT)/$(ATH_ANDROID_SRC_BASE)
 	$(ACP) -fpt $(ATH_ANDROID_SRC_BASE)/compat/compat.ko $(TARGET_OUT)/lib/modules/
 	$(ACP) -fpt $(ATH_ANDROID_SRC_BASE)/net/wireless/cfg80211.ko $(TARGET_OUT)/lib/modules/
 
diff --git a/wlan/atheros/compat-wireless/Makefile b/wlan/atheros/compat-wireless/Makefile
index 47af45f..8c571ca 100644
--- a/wlan/atheros/compat-wireless/Makefile
+++ b/wlan/atheros/compat-wireless/Makefile
@@ -13,11 +13,18 @@ OLD_IWL=$(shell $(MODPROBE) -l iwl4965)
 
 DESTDIR?=
 
+export COMPAT_ROOT := $(COMPAT_CURDIR)
+ccflags-y += \
+        -DCOMPAT_BASE_TREE="\"$(shell cat $(COMPAT_ROOT)/compat_base_tree)\"" \
+        -DCOMPAT_BASE_TREE_VERSION="\"$(shell cat $(COMPAT_ROOT)/compat_base_tree_version)\"" \
+        -DCOMPAT_PROJECT="\"Compat-wireless\"" \
+        -DCOMPAT_VERSION="\"$(shell cat $(COMPAT_ROOT)/compat_version)\""
+
 ifneq ($(KERNELRELEASE),)
 
 NOSTDINC_FLAGS := -I$(M)/include/ \
 	-include $(M)/include/linux/compat-2.6.h \
-	$(CFLAGS)
+	$(ccflags-y)
 
 obj-y := compat/
 
@@ -38,12 +45,6 @@ endif
 
 else
 
-export COMPAT_ROOT := $(shell pwd)
-CFLAGS += \
-        -DCOMPAT_BASE_TREE="\"$(shell cat compat_base_tree)\"" \
-        -DCOMPAT_BASE_TREE_VERSION="\"$(shell cat compat_base_tree_version)\"" \
-        -DCOMPAT_PROJECT="\"Compat-wireless\"" \
-        -DCOMPAT_VERSION="\"$(shell cat compat_version)\""
 
 # These exported as they are used by the scripts
 # to check config and compat autoconf
-- 
1.8.0

